src/ripple/ripple.js

```diff
@@ -109,25 +110,52 @@
       }
       this.setRippleXY(x, y);
       this.setRippleStyles(true);
-      requestAnimationFrame(this.animFrameHandler.bind(this));
+
+      if (this.timeout) {
+        window.clearTimeout(this.timeout);
+      }
+
+      // We've just clicked, clear past upHandler_ call record
+      this.rippleElement_.mouseUp = false;
+      // Indicate to upHandler_ that ripple hasn't run for this.Constant_.DURATION
+      this.rippleElement_.animationCompleted = false;
+
+      window.requestAnimationFrame(function() {
+        this.animFrameHandler();
+
+        // Allow the animation to complete, even for tap events, which may
+        // trigger multiple mouseup with final event.detail == 1
+        // Ex: mac trackpad tap triggers 2 consecutive event.detail == 0 events
+        // followed by an event.detail=1 mouseup event
+        // we set the animation timeout inside the frame request to
+        // prevent animation flashing during long ui-blocking operations
+        this.timeout = window.setTimeout(function() {
```

Comment:
Probably `setTimeout` instead of `window.setTimeout`, similarly to `requestAnimationFrame`.